### ISSUE TITLE
Error:: Tracking Module:: t must be passed in for this to work

### ISSUE THREAD

**Issue by srajangoyal** [avstack-lab/lib-avstack-core#5]
For now I tried `basic-box-tracker`, to track 3dobjects.
Could you please check?

Partial traceback:
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-core/avstack/ego/vehicle.py", line 701, in _tick_modules
    tracks_camera = self.tracking["camera"](
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-core/avstack/modules/tracking/base.py", line 131, in __call__
    raise RuntimeError("t must be passed in for this to work")
RuntimeError: t must be passed in for this to work
Done!

**Comment by roshambo919** [comment]
The tracking API is a bit clunky right now, but that error message basically means you need to pass arguments in as keyword arguments.

e.g., like this:
```
tracks_3d = self.tracking(t=timestamp, detections_nd=dets_3d, frame=frame, identifier="tracker-0")
```
Also, are you sure you want to be using 3d tracking for a "camera" tracker?

**Comment by srajangoyal** [comment]
I tried this before, but the order of arguments was different. Thank you for this info.
There is another error using camera for 3d object tracking. Is this expected? Maybe due to the frame rate inconsistent with the simulation?l

Detected 2 objects with our camera     
WARNING:root:'NoneType' object has no attribute 'data'
Traceback (most recent call last):
  File "exec_standard.py", line 81, in main
    done = carla_manager.tick()
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-api/avapi/carla/simulator/manager.py", line 88, in tick
    done = self.ego.tick(self.t_elapsed, self.frame, self.infrastructure)
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-api/avapi/carla/simulator/ego.py", line 298, in tick
    ctrl = self.algorithms.tick(
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-core/avstack/ego/vehicle.py", line 37, in tick
    control = self._tick_modules(
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-core/avstack/ego/vehicle.py", line 182, in _tick_modules
    dets_3d_cam = self.perception["object_3d_cam"](
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-core/avstack/modules/perception/base.py", line 34, in __call__
    detections = self._execute(
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-core/avstack/modules/perception/object3d.py", line 95, in _execute
    result_ = self.run_mm_inference(
  File "/home/srajan/carla-sandbox/submodules/lib-avstack-core/avstack/modules/perception/object3d.py", line 131, in run_mm_inference
    imwrite(data_file, data.data)
AttributeError: 'NoneType' object has no attribute 'data'
Done!


For sure, it's not a good choice. But since lidar is not yet working ([other issue](https://github.com/avstack-lab/lib-avstack-core/issues/4)), and `AdaptiveCruiseControl` takes only 3d_objects as input (`modules.planning.components.get_object_to_follow()`), I had to try.

**Comment by roshambo919** [comment]
FYI I simplified the tracking interface in the latest commit 4e22aab7c7f97ed51b92025c939bb8a31f624646.

Regarding the second issue, that is also a bug that I've patched (I think) in the latest commit 4e22aab7c7f97ed51b92025c939bb8a31f624646.

HOWEVER, currently, I had to revert the LiDAR based perception. I need to retrain a model on the carla data or need to get something working that isn't nuscenes (because of the number of columns). Stay tuned for that. I opened another issue here: #7
